src/Draco.Compiler/Internal/Binding/Binder_Lookup.cs

```diff
@@ -79,6 +79,12 @@ private LookupResult LookupInternal(string name, Predicate<Symbol> allowSymbol,
 
         // Set up state
         var flags = LookupFlags.None;
+        if (this.Compilation.Flags.HasFlag(Api.CompilationFlags.ScriptingMode))
+        {
+            // In scripting mode, we allow globals to be looked up immediately without crossing
+            // a function boundary
+            flags |= LookupFlags.AllowGlobals;
+        }
```

Comment:
This screws up everything else, not just top-level statements. Solve with binder specialization instead.